### PR TITLE
feat: admin form block

### PR DESCRIPTION
**Pull request by csiyang**
# Description

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 5501618</samp>

This pull request adds support for creating and configuring FormBlocks using Formium in the journeys-admin app. It updates the schema types, queries, and mutations for the FormBlock type and its related fields. It also adds custom components for editing the FormBlock attributes, such as the credentials and the action. It modifies the .vscode/settings.json file to improve the developer experience with eslint.

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/35391169/todos/6813111451)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

# Walkthrough

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 5501618</samp>

*  Update the schema for the GridContainerBlock type and fix the type errors in the generated files ([link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-d6685198bb51cd1cec3d87ff74ffbc1b2ec1fc9c5b4761b68b5ff63a6cd12563L12-R13), [link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-d6685198bb51cd1cec3d87ff74ffbc1b2ec1fc9c5b4761b68b5ff63a6cd12563L560-R617), [link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-9fecf255f6a72895588618cbc843b8a568d1f488e250d433135d9762c70dc518L26-R27), [link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-9fecf255f6a72895588618cbc843b8a568d1f488e250d433135d9762c70dc518L574-R631), [link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-1ec4c02e97a38e8aed295fe74ec2c3ac54754837958d07baf3587e7f8e9cf427L26-R27), [link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-1ec4c02e97a38e8aed295fe74ec2c3ac54754837958d07baf3587e7f8e9cf427L574-R631), [link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-8e469721bafdbc2073a68320235979721c971a1404c2bbe6f0ec7a2b42f2fb5cL26-R27), [link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-8e469721bafdbc2073a68320235979721c971a1404c2bbe6f0ec7a2b42f2fb5cL574-R631), [link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-a015b5ceb6a29272281dee2bd576cb9baad85ff58bc0981d596dbc7b84d72665L26-R27), [link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-a015b5ceb6a29272281dee2bd576cb9baad85ff58bc0981d596dbc7b84d72665L574-R631))
*  Add the interfaces for the different action types that a FormBlock can have and use them in the generated files ([link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-d6685198bb51cd1cec3d87ff74ffbc1b2ec1fc9c5b4761b68b5ff63a6cd12563R114-R170), [link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-9fecf255f6a72895588618cbc843b8a568d1f488e250d433135d9762c70dc518R128-R184), [link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-1ec4c02e97a38e8aed295fe74ec2c3ac54754837958d07baf3587e7f8e9cf427R128-R184), [link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-8e469721bafdbc2073a68320235979721c971a1404c2bbe6f0ec7a2b42f2fb5cR128-R184), [link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-a015b5ceb6a29272281dee2bd576cb9baad85ff58bc0981d596dbc7b84d72665R128-R184))
*  Add the types for the input fields of the FormBlockCreate and FormBlockUpdate mutations and use them in the globalTypes file ([link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-51d1d0a9862e904e37cd586635df16bbf3a45505b21cbf0908386407b7ec27e8R301-R312))
*  Add the FormBlockCreate mutation operation and use it in the FormBlockCreate file ([link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-d791de19d91c20703c64f118aad3cd8078536c3ccdd939d8ab02c0924139aa06R1-R75))
*  Add the FormBlockUpdateCredentials mutation operation and use it in the FormBlockUpdateCredentials file ([link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-60ec0f8da5bdd17ca1f55feb8cba19c5e80d367d50a5da90fd4fb69ce92b36f2R1-R53))
*  Add the FormFields fragment and use it in the FormFields file ([link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-2f50ffa2101c2b9d64688638c79b2905a89c181de119d2b6dae34a7172a44c50R1-R65))
*  Add the GetFormBlock query operation and use it in the GetFormBlock file ([link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-a8e7ffe706e495756d6396bc681708eef080bdf414df5dd69b9fcef15b14dd37R1-R56))
*  Modify the editor.codeActionsOnSave setting to use the "explicit" value in the .vscode/settings.json file ([link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-a5de3e5871ffcc383a2294845bd3df25d3eeff6c29ad46e3a396577c413bf357L76-R76))
*  Use the onboardingFormiumClient instead of the formiumClient for the onboarding form in the onboarding-form file ([link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-4c59beef32f5a1c81172fc63a0abcd2a1f7d73962c38388d409fb3c17bcb60caL12-R12), [link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-4c59beef32f5a1c81172fc63a0abcd2a1f7d73962c38388d409fb3c17bcb60caL52-R52))
*  Use the Form component for rendering the attributes of a FormBlock in the Attributes file ([link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-7b2ae8d25a5b91a8837f7e1893c697072da269a876afc435eea0649b57b06083R13), [link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-7b2ae8d25a5b91a8837f7e1893c697072da269a876afc435eea0649b57b06083R49-R51))
*  Define and export the ApiTokenTextField component for editing the apiToken field of a FormBlock in the ApiTokenTextField files ([link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-6ac2cc2bec7068dd88f649b7fdf800cb32b537f55868dead2708743be481cdefR1-R86), [link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-d08f9f42821a6acdda9183b84d39ec6a3866f57799818afef8a6c6e758e7bceeR1))
*  Define the Credentials component for rendering the credentials fields of a FormBlock in the Credentials file ([link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-76d09fe3d426e3450b4be23c1fad59fa15c79c0a4b032ea3b15eaaf06457b976R1-R81))
*  Define and export the FormSlugSelect component for editing the formSlug field of a FormBlock in the FormSlugSelect files ([link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-d848fd73a657162890f970a21ca597db96d128fb2e3642c0f8e450053c7e7568R1-R77), [link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-1e662e34ce31d188db50b4f74f7e5c7f36b7b1e9f235dccf7b3be0b7f393e7c7R1))
*  Define and export the ProjectIdSelect component for editing the projectId field of a FormBlock in the ProjectIdSelect files ([link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-2f5b621b8d74ee6a95c7f03707b0efadadbc64886419eb089f889a747735bb84R1-R77))
*  Use the FormBlock type for the selectedBlock variable when the selected block is a FormBlock in the Action file ([link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-5a35a228f99f223cb1f367e925c18fcfb801da1fc265692c8bda5c2fc742ea02R18), [link](https://github.com/JesusFilm/core/pull/2131/files?diff=unified&w=0#diff-5a35a228f99f223cb1f367e925c18fcfb801da1fc265692c8bda5c2fc742ea02L79-R84))

### CODE REVIEW

File: apps/journeys-admin/src/components/Editor/ControlPanel/BlocksTab/BlocksTab.tsx [hunk 1]
```diff
@@ -30,6 +31,8 @@ export function BlocksTab(): ReactElement {
         <NewTextResponseButton />
         <NewSignUpButton />
         <NewButtonButton />
+        {/* TODO: comment this out before merging to PROD until its put behind a flag */}
+        <NewFormButton />
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/ControlPanel/BlocksTab/BlocksTab.tsx:35]: rember to comment out before merging

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Form/Credentials/ApiTokenTextField/ApiTokenTextField.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    }
+  }
+`
+
+interface ApiTokenTextFieldProps {
+  id?: string
+}
+
+export const placeHolderToken =
+  'thisIsAFakeApiTokenTheReaOneIsNeverShowAgainInTheFrontEnd'
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Form/Credentials/ApiTokenTextField/ApiTokenTextField.tsx:33]: what happens if this is a valid API token? lol.

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Form/Credentials/FormSlugSelect/FormSlugSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      <Select
+        onChange={handleChange}
+        value={loading ? '' : currentFormSlug ?? 'none'}
+        IconComponent={ChevronDownIcon}
+      >
+        <MenuItem key="form-formSlug-none" value="none">
+          {t('None')}
+        </MenuItem>
+        {forms?.map((form) => (
+          <MenuItem key={`form-formSlug-${form.slug}`} value={form.slug}>
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Form/Credentials/FormSlugSelect/FormSlugSelect.tsx:59]:
```suggestion
          <MenuItem key={form.slug} value={form.slug}>
```

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Form/Credentials/ProjectIdSelect/ProjectIdSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      <Select
+        onChange={handleChange}
+        value={loading ? '' : currentProjectId ?? 'none'}
+        IconComponent={ChevronDownIcon}
+      >
+        <MenuItem key="form-projectId-none" value="none">
+          {t('None')}
+        </MenuItem>
+        {projects?.map((project) => (
+          <MenuItem key={`form-projectId-${project.id}`} value={project.id}>
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Form/Credentials/ProjectIdSelect/ProjectIdSelect.tsx:59]:
```suggestion
          <MenuItem key={project.id} value={project.id}>
```

File: libs/journeys/ui/src/components/Form/Form.spec.tsx [hunk 5]
```diff
@@ -7,11 +7,6 @@ import { FormFields } from './__generated__/FormFields'
...
 import { Form } from '.'
 
-jest.mock('next-firebase-auth', () => ({
-  __esModule: true,
-  useUser: jest.fn(() => ({ id: 'userId', name: 'userName' }))
-}))
-
 jest.mock('react-i18next', () => ({
   __esModule: true,
   useTranslation: () => {
```
Comment (tataihono) [inline libs/journeys/ui/src/components/Form/Form.spec.tsx:12]: drop

**Review by tataihono — CHANGES_REQUESTED** [review]

**Review by tataihono — DISMISSED** [review]

**Review by tataihono — APPROVED** [review]

**Pull request merged**